app/src/main/java/com/example/pix/home/fragments/ChatsFragment.java

```diff
@@ -99,9 +141,11 @@ public void onLoadMore(int page, int totalItemsCount, RecyclerView unusedView) {
                             toRemove.add(c.getObjectId());
                         }
                         // Delete them
-                        for (Chat c : chats){
-                            if (toRemove.contains(c.getObjectId())) {
-                                chats.remove(c);
+                        // The short-hand for-loop gave concurrent complications at times
+                        for (int i = 0; i < chats.size(); i++) {
+                            if (toRemove.contains(chats.get(i).getObjectId())) {
+                                chats.remove(i);
+                                i--;
```

Comment:
What if i is 0 here? Also why are you decrementing the i here? And if you remove something from the list, the value from `chats.size()` might not be valid anymore. You might want to rewrite this part.

Reply:
I'm looping through the chats, and if I remove say Chat with index 4, the Chat that was index 5 will now be index 4, so I decrement i to 3 before the for loop increments it back to 4 so I may recheck index 4 (Which is the old index 5). I think it works well enough, but if it seems too strange then I can change it.